src/coreneuron/CMakeLists.txt

```diff
@@ -607,11 +607,6 @@ target_include_directories(
   coreneuron-core SYSTEM PRIVATE ${CLI11_HEADER_DIRECTORY}
                                  ${PROJECT_SOURCE_DIR}/external/Random123/include)
 
-# See: https://en.cppreference.com/w/cpp/filesystem#Notes
-if(CMAKE_CXX_COMPILER_IS_GCC AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1)
```

Comment:
@alexsavulescu : asking for clarity - does our build system error if the user tries to use older GCC? (if not, should we turn such a check into emitting a CMake error then?)